Dockerfile.build

```diff
@@ -5,17 +5,23 @@ ARG BAZELISK_VERSION=1.8.0
 ARG PI_COMMIT=a5fd855d4b3293e23816ef6154e83dc6621aed6a
 ARG BMV2_COMMIT=b0fb01ecacbf3a7d7f0c01e2f149b0c6a957f779
 ARG JDK_URL=https://mirror.bazel.build/openjdk/azul-zulu11.29.3-ca-jdk11.0.2/zulu11.29.3-ca-jdk11.0.2-linux_x64.tar.gz
-ARG LLVM_REPO_NAME="deb http://apt.llvm.org/stretch/  llvm-toolchain-stretch-11 main"
+ARG LLVM_REPO_NAME="deb http://apt.llvm.org/buster/  llvm-toolchain-buster-11 main"
+ARG SONIC_KERNEL_HEADER_URL_COMMON="https://sonic-build.azurewebsites.net/api/sonic/artifacts?branchName=202106&platform=barefoot&buildId=116439&target=target%2Fdebs%2Fbuster%2Flinux-headers-4.19.0-12-2-common_4.19.152-1_all.deb"
+ARG SONIC_KERNEL_HEADER_URL_AMD64="https://sonic-build.azurewebsites.net/api/sonic/artifacts?branchName=202106&platform=barefoot&buildId=116439&target=target%2Fdebs%2Fbuster%2Flinux-headers-4.19.0-12-2-amd64_4.19.152-1_amd64.deb"
```

Comment:
@bocon13 this kinda goes against our plan of packaging and hosting this externally. Do you see any problems with this approach?

To me this here made sense because:
- ensures the header files are there, no need to download stuff repeatedly
- version controlled
- easy to update

Are we concerned about having to build/pull a new builder image on kernel updates?

Reply:
We decided to not have kernel sources inside the docker image. Instead they go into the `sonic-base-image` repo and will be downloaded when needed.